nixos/modules/services/web-apps/immich-kiosk.nix

```
@@ -0,0 +1,180 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.immich-kiosk;
+  format = pkgs.formats.yaml { };
+  inherit (lib)
+    types
+    mkIf
+    mkOption
+    mkEnableOption
+    mkPackageOption
+    ;
+in
+{
+  meta.maintainers = with lib.maintainers; [ tlvince ];
+
+  options.services.immich-kiosk = {
+    enable = mkEnableOption "Immich Kiosk slideshow service";
+
+    package = mkPackageOption pkgs "immich-kiosk" { };
+
+    immichUrl = mkOption {
+      type = types.str;
+      example = "https://immich.example.com";
+      description = ''
+        URL of the Immich instance. Must include a port if one is needed.
+      '';
+    };
+
+    immichApiKeyFile = mkOption {
+      type = lib.types.nullOr (
+        lib.types.pathWith {
+          inStore = false;
+          absolute = true;
+        }
+      );
+      default = null;
+      example = "/run/secrets/immich-kiosk-api-key";
+      description = ''
+        Path to a file containing the Immich API key.
+        The file should contain only the API key with no trailing newline.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 3000;
+      description = ''
+        Port on which immich-kiosk will listen.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to open the firewall for the immich-kiosk port.
+      '';
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = format.type;
+      };
+      default = { };
+      description = ''
+        Configuration for immich-kiosk. See
+        <https://github.com/damongolding/immich-kiosk#configuration>
+        for available options.
+      '';
+      example = lib.literalExpression ''
+        {
+          albums = [
+            "4fa933cf-051f-4621-9ac7-8d06776c261c"
+            "6466548c-4995-4fb5-ab1f-f63cc9ff3e5f"
+          ];
+          duration = 30;
+          layout = "splitview";
+          disable_ui = true;
+        }
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      example = "/run/secrets/immich-kiosk-env";
+      description = ''
+        Environment file as defined in {manpage}`systemd.exec(5)`.
+
+        All KIOSK_* environment variables are supported.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.immichApiKeyFile != null;
+        message = "services.immich-kiosk.immichApiKeyFile must be set";
+      }
+    ];
+
+    systemd.services.immich-kiosk = {
+      description = "Immich Kiosk slideshow service";
+      after = [ "immich-server.service" ];
```

This should probably be conditional, as people could be using this with a remote Immich instance or with a Docker deployment.


```suggestion
      after = lib.mkIf config.services.immich.enable [ "immich-server.service" ];
```